### ISSUE TITLE
Failing to play a playlist takes too long

### ISSUE THREAD

**Issue by arcnmx** [arcnmx/MPD#5]
When playing the result of `mpc add https://youtube.com/playlist?list=xxx`, there is no single song to play, so as of #4 we bail out because youtube-dl returns a playlist json instead. Currently, Invoke calls `youtube-dl --no-playlist URL`. When it falls back to returning a playlist, it operates in "full" mode by default, which takes forever and is bad.

The solution here is probably to change invoke arguments to use `--flat-playlist --no-playlist` instead when we want to play a single song URL - even though we don't actually want any playlist data. It'll return faster and fail faster. This should work fine, since youtube-dl still obeys `--no-playlist` and returns a single song result even when passed a hybrid song+playlist URL. In other words, every youtube-dl command (whether for input or playlist) should probably just include `--flat-playlist` - the plugin doesn't even currently expose full playlists anyway. I'm not sure if there exist any sites where it's useful anyway?

**Comment by Rio6** [comment]
Moving conversion from #4 here.

> I think we can keep FULL, it's not much of the code anyways.

Actually, just looked at the code. It seems like it'd be easier to add `--flat-playlist` on both FLAT and SINGLE if we remove FULL, since `--flat-platlist` can just be one of the universal options like this:
```diff
diff --git a/src/lib/ytdl/Invoke.cxx b/src/lib/ytdl/Invoke.cxx
index 175c8ed35..958ad2249 100644
--- a/src/lib/ytdl/Invoke.cxx
+++ b/src/lib/ytdl/Invoke.cxx
@@ -64,15 +64,12 @@ YtdlProcess::Invoke(Yajl::Handle &handle, const char *url, PlaylistMode mode)
                                playlist_flag = "--no-playlist";
                                break;
                        case PlaylistMode::FLAT:
-                               playlist_flag = "--flat-playlist";
-                               break;
-                       case PlaylistMode::FULL:
                                playlist_flag = "--yes-playlist";
                                break;
                }
 
                if (execlp("youtube-dl", "youtube-dl",
-                       "-Jf", "bestaudio/best", playlist_flag, url, nullptr) < 0)
+                       "-Jf", "bestaudio/best", "--flat-playlist", playlist_flag, url, nullptr) < 0)
                {
                        _exit(EXIT_FAILURE);
                }

```

What do you think?

**Comment by arcnmx** [comment]
You could probably tack it onto the end after the url and set it to null if it's unneeded? but... yeah it's easier and cleaner to just remove FULL since it doesn't change the number of arguments. I'm fine with removing it for now, FULL is dead code and it can be brought back if we find an actual problem with using flat mode in some scenario later on.

**Comment by arcnmx** [comment]
Fixed in #10
